src/main/java/org/broadinstitute/hellbender/tools/copynumber/gcnv/GenerateVCFFromPosteriors.java

```diff
@@ -0,0 +1,224 @@
+        }
+
+        final SimpleIntervalCollection firstChunkSimpleIntervalCollection =
+                new SimpleIntervalCollection(new File(getIntervalFileFromChunkDirectory(orderedChunkDirectoryList.get(0).getAbsolutePath()).getAbsolutePath()));
+        final SAMSequenceDictionary samSequenceDictionary = firstChunkSimpleIntervalCollection.getMetadata().getSequenceDictionary();
+        Triple<List<CopyNumberPosteriorLocatableRecord>, IntegerCopyNumberStateCollection, String> firstCopyNumberPosteriorFileInfo =
+                readChunkedPosteriorFileFromDirectory(orderedChunkDirectoryList.get(0), samSequenceDictionary);
+        final IntegerCopyNumberStateCollection copyNumberStateCollection = firstCopyNumberPosteriorFileInfo.getMiddle();
+        sampleName = firstCopyNumberPosteriorFileInfo.getRight();
+
+        if (firstCopyNumberPosteriorFileInfo.getMiddle().size() < 3) {
+            throw new UserException.BadInput("There should be at least 3 copy number states");
+        }
+
+        final VariantContextWriter outputWriter = GATKVariantContextUtils.createVCFWriter(outputFile, null, false);
+        final GCNVPostProcessor gcnvPostProcessor = new GCNVPostProcessor(outputWriter, copyNumberStateCollection,
+                sampleName, samSequenceDictionary);
+
+        //TODO pass the command line invocation string to the header composer method
+        gcnvPostProcessor.composeVariantContextHeader(null);
+        int currentChunk = 0;
+
+        for(File chunkRootDirectory: orderedChunkDirectoryList) {
+            currentChunk++;
+            logger.info(String.format("Analyzing copy number posterior chunk %d", currentChunk));
+
+            Triple<List<CopyNumberPosteriorLocatableRecord>, IntegerCopyNumberStateCollection, String> chunkedCopyNumberPosteriorFileInfo =
+            readChunkedPosteriorFileFromDirectory(chunkRootDirectory, samSequenceDictionary);
+            //check that the copy number state collection is equivalent across all chunks
+            if (!copyNumberStateCollection.equals(chunkedCopyNumberPosteriorFileInfo.getMiddle())) {
+                throw new UserException.BadInput("Copy number collection differs across chunked posterior outputs");
+            }
+            //check that the sample name is consistent across all chunks
+            if (!chunkedCopyNumberPosteriorFileInfo.getRight().equals(sampleName)) {
+                throw new UserException.BadInput("Sample name differs across chunked posterior outputs");
+            }
+
+            final List<CopyNumberPosteriorLocatableRecord> copyNumberPosteriorLocatableRecordsList =
+                    chunkedCopyNumberPosteriorFileInfo.getLeft();
+
+            gcnvPostProcessor.writeChunkedVariantContext(copyNumberPosteriorLocatableRecordsList, variantPrefix);
+            currentChunk++;
+        }
+        outputWriter.close();
+        return;
+    }
+
+    /**
+     * Read in a single posterior chunked file
```

Comment:
`Reads and processes a {@link GermlineCNVCaller} analysis chunk for a single sample.`
Also, add a description for the return type.

Reply:
Done